### PR TITLE
Feat : Playlist added

### PR DESCRIPTION
**Pull request by yashodaAgarwal**

### CODE REVIEW

File: src/components/Playlist/playlistModal.css [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    display: flex;
+    align-items: center;
+    gap: 10px;
+    position: relative;
+  }
+  
+  .add-playlist {
+    position: absolute;
+    right: 8px;
+    font-size: 26px;
```
Comment (Ananyamadhu08) [inline src/components/Playlist/playlistModal.css:62]: can use rem for higher values

File: src/components/Playlist/playlistcard.css [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  .playlist-details {
+    padding: 7px 10px;
+    display: flex;
+    align-items: flex-end;
+    justify-content: space-between;
+  }
+  
+  .delete-icon {
+    font-size: 24px;
+    margin-bottom: 20px;
```
Comment (Ananyamadhu08) [inline src/components/Playlist/playlistcard.css:25]: can use rem here

File: src/context/index.js [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import { AuthProvider, useAuth } from "./AuthContext";
 import { ServerProvider, useServer } from "./Categorycontext";
```
Comment (Ananyamadhu08) [inline src/context/index.js:2]: you can use barrel imports to maintain readability.

**Review by Ananyamadhu08 — APPROVED** [review]
Please look into the suggested changes, other than that everything looks good. Nice variable naming. You can go ahead and merge.

File: src/App.js [hunk 4]
```diff
@@ -21,6 +23,8 @@ const  App = () => {
       <Route path="/login" element={<Loginpage/>}/>
       <Route path="/signup" element={<Signup/>}/>
       <Route path="/watchlater" element={<RequiresAuth><WatchLaterPage/></RequiresAuth>}/>
+      <Route path="/playlist/:playlistId" element={<RequiresAuth><PlaylistPage/></RequiresAuth>}/>
+      <Route path="/playlist" element={<RequiresAuth><Playlist/></RequiresAuth>}/>
       <Route path="/like" element={<RequiresAuth><LikePage/></RequiresAuth>}/>
       <Route path="/SingleVideo" element={<SingleVideoPage/>}/>
       <Route path="/history" element={<RequiresAuth><HistoryPage/></RequiresAuth>}/>
```
Comment (rahulyadav139) [inline src/App.js:30]: suggestion: you can use the outlet here which is provided by react-router. Try that it is great way to protect routes.

**Review by rahulyadav139 — APPROVED** [review]
Appreciation: everything is looking great. Codes are readable. Great work.

**Pull request merged**